spec/src/main/asciidoc/microprofile-lra-spec.adoc

```diff
@@ -731,6 +732,107 @@ Only when the implementation is certain that participant has finished
 will it tell it that it is okay to release any resources it associated
 with the LRA.
 
+[[reactive-support]]
+==== Reactive Support
+
+Implementations are expected to operate correctly when services use the
+asychronous and reactive features provided by JAX-RS.
+
+It has already been noted that
+<<async-compensators,participant completion and compensation callbacks>>
+can execute asynchronously but the same must also be true for the business
+methods that execute in the context of an LRA. It is the responsibility
+of the implementation to ensure that JAX-RS asynchronous features
+continue to behave in the presence of these LRAs. The following
+example shows a resource invocation that runs in the context of a
+long running action and uses a Java 8 completion stage to return
+an asynchronous response:
+
+[source,java]
+----
+    @LRA(value = LRA.Type.REQUIRED, // the method must run with an LRA
+         end = true) // the LRA must end when the method completes
+    @Path("completion-stage-lra")
+    public CompletionStage<Response> asyncInvocationWithLRA(
+        @HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) { // the id of LRA
+
+        ExecutorService es = Executors.newSingleThreadExecutor();
```

Comment:
I'm not sure it's a good idea to show to the user the creation of a thread pool in a JAX-RS method. 

What about:

```
service.doSomethingAsynchrnously() // returns a CompletationStage<Void>
    .thenApply(x -> Response.ok().entity(lraId).build())
```

Reply:
It would also remove the confusing thread pool shutdown.

Reply:
@xstefank preferred that option too